Migrating ERC-20 Files to docs-app

Buddhika-Pathirathna opened the pull request.

Buddhika-Pathirathna commented:
> The content isn't showing up on the local run, so I cannot review without this fix. Please check the sidebar.config.js, it needs some fixes and the label needs to change to **ERC-20** all-caps.

@GloreenS as we discussed yesterday, I checked this again cloning the branch to a separate folder, there also the content was visible for me. @ipopescu, could you please help us confirm this - Gloreen was unable to load the content and guess it could be some issue in sidebar.config.js,  I can't see it since it's working fine in my machine, can you please try to load it in your machine and check?

ipopescu commented:
> ain cloning the branch to a separate folder, there also the content was visible for me. @ipopescu, could you please help us confirm this - Gloreen was unable to load the content and guess it could be some issue in sidebar.config.js, I can't see it since it's working fine in my machine, can you please try to load it in your machine and check?

@Buddhika-Pathirathna I am not able to load the docs-app locally. When I issue the command "yarn run start" I get the following error: 
```
- workflow/undelegate
    at Object.checkSidebarsDocIds (/Users/iulia/casper/docusaurus/docs-app/node_modules/@docusaurus/plugin-content-docs/lib/sidebars.js:339:19)
    at doLoadVersion (/Users/iulia/casper/docusaurus/docs-app/node_modules/@docusaurus/plugin-content-docs/lib/index.js:117:31)
    at async loadVersion (/Users/iulia/casper/docusaurus/docs-app/node_modules/@docusaurus/plugin-content-docs/lib/index.js:166:28)
    at async Promise.all (index 0)
    at async Object.loadContent (/Users/iulia/casper/docusaurus/docs-app/node_modules/@docusaurus/plugin-content-docs/lib/index.js:174:33)
    at async /Users/iulia/casper/docusaurus/docs-app/node_modules/@docusaurus/core/lib/server/plugins/index.js:58:46
    at async Promise.all (index 0)
    at async Object.loadPlugins (/Users/iulia/casper/docusaurus/docs-app/node_modules/@docusaurus/core/lib/server/plugins/index.js:57:27)
    at async Object.load (/Users/iulia/casper/docusaurus/docs-app/node_modules/@docusaurus/core/lib/server/index.js:186:82)
    at async start (/Users/iulia/casper/docusaurus/docs-app/node_modules/@docusaurus/core/lib/commands/start.js:44:19)
error Command failed with exit code 1.
```

ipopescu commented on source/docs/casper/workflow/erc-20/introduction.md:
```diff
@@ -0,0 +1,33 @@
+
+The four users are created in [Casper Integration](https://integration.cspr.live/) live site and following are the accounts created to demonstrate the use of ERC20 contract:
+
+-   [User A](https://integration.cspr.live/account/01f2dfc09a94ef7bce440f93a1bb6f17fdac0c913549927d452e7e91a376e9d20d)
+
+-   [User B](https://integration.cspr.live/account/015d4e20b5f7f687be80aed6e20960898b02c7549cc49ddf583224ecd894eca375)
+
+-   [User C](https://integration.cspr.live/account/0101fe69ae2012358e5ce8e8b39661d45d225251c4f19ebb7fc74b057637e65aa4)
+
+-   [User D](https://integration.cspr.live/account/0171bd7bac58780ce950007de575a472bcb30457e7b68427a6ed466568d71db1d6)
```
Please remove the extra space between bullets.

ipopescu commented on source/docs/casper/workflow/erc-20/introduction.md:
```diff
@@ -0,0 +1,33 @@
+# Introduction
```
Please use the content as it is here: https://github.com/casper-network/docs/tree/master/workflow/erc-20. I noticed that the content does not match. Further improvements or updates can be added as a separate PR.

ipopescu commented on source/docs/casper/workflow/erc-20/setup.md:
```diff
@@ -0,0 +1,70 @@
+# Setup
+
+To initiate the set up:
```
Please use the same introduction found here: https://docs.casperlabs.io/en/latest/workflow/erc-20/setup.html

ipopescu commented on source/docs/casper/workflow/erc-20/transfers.md:
```diff
@@ -0,0 +1,251 @@
+    --secret-key ~/casper/ibm_demo/user_a/secret_key.pem
+    --session-hash hash-b568f50a64acc8bbe43462ffe243849a88111060b228dacb8f08d42e26985180
+    --session-entry-point "transfer"
+    --session-arg "recipient:key='account-hash-9f81014b9c7406c531ebf0477132283f4eb59143d7903a2fae54358b26cea44b" --session-arg "amount:u256='50'"
```
Move the second session-arg on a new line

ipopescu commented on source/docs/casper/workflow/erc-20/approvals.md:
````diff
@@ -0,0 +1,271 @@
+
+```
+    casper-client put-deploy -n http://3.143.158.19:7777 \
+    --chain-name integration-test --secret-key ~/casper/ibm_demo/user_c/secret_key.pem \
````
Move the secret key on a new line.

ipopescu commented on source/docs/casper/workflow/erc-20/transfers.md:
```diff
@@ -0,0 +1,251 @@
+    --session-package-name "erc20_test_call"
+    --session-entry-point "check_balance_of"
+    --session-arg "token_contract:account_hash='account-hash-b568f50a64acc8bbe43462ffe243849a88111060b228dacb8f08d42e26985180'"
+    --session-arg "address:key='account-hash-303c0f8208220fe9a4de40e1ada1d35fdd6c678877908f01fddb2a56502d67fd'" --chain-name integration-test
```
Move chain-name on a new line.

ipopescu commented on source/docs/casper/workflow/erc-20/transfers.md:
```diff
@@ -0,0 +1,251 @@
+    --secret-key ~/casper/ibm_demo/user_b/secret_key.pem
+    --session-hash hash-b568f50a64acc8bbe43462ffe243849a88111060b228dacb8f08d42e26985180
+    --session-entry-point "transfer"
+    --session-arg "recipient:key='account-hash-89422a0f291a83496e644cf02d2e3f9d6cbc5f7c877b6ba9f4ddfab8a84c2670'" --session-arg "amount:u256='20'"
```
Move the second session-arg on a new line.

ipopescu commented on source/docs/casper/workflow/erc-20/transfers.md:
```diff
@@ -0,0 +1,251 @@
+    --session-package-name "erc20_test_call"
+    --session-entry-point "check_balance_of"
+    --session-arg "token_contract:account_hash='account-hash-b568f50a64acc8bbe43462ffe243849a88111060b228dacb8f08d42e26985180'"
+    --session-arg "address:key='account-hash-303c0f8208220fe9a4de40e1ada1d35fdd6c678877908f01fddb2a56502d67fd'" --chain-name integration-test
```
Move chain-name on a new line.

ipopescu commented on source/docs/casper/workflow/erc-20/transfers.md:
```diff
@@ -0,0 +1,251 @@
+    --session-package-name "erc20_test_call"
+    --session-entry-point "check_balance_of"
+    --session-arg "token_contract:account_hash='account-hash-b568f50a64acc8bbe43462ffe243849a88111060b228dacb8f08d42e26985180'"
+    --session-arg "address:key='account-hash-9f81014b9c7406c531ebf0477132283f4eb59143d7903a2fae54358b26cea44b'" --chain-name integration-test
```
Move chain-name on a new line.

ipopescu commented on source/docs/casper/workflow/erc-20/transfers.md:
```diff
@@ -0,0 +1,251 @@
+    --session-package-name "erc20_test_call"
+    --session-entry-point "check_balance_of"
+    --session-arg "token_contract:account_hash='account-hash-b568f50a64acc8bbe43462ffe243849a88111060b228dacb8f08d42e26985180'"
+    --session-arg "address:key='account-hash-89422a0f291a83496e644cf02d2e3f9d6cbc5f7c877b6ba9f4ddfab8a84c2670'" --chain-name integration-test
```
Move chain-name on a new line.

ipopescu commented on source/docs/casper/workflow/erc-20/transfers.md:
```diff
@@ -0,0 +1,251 @@
+    --chain-name integration-test
+    --secret-key ~/casper/ibm_demo/user_a/secret_key.pem
+    --session-hash hash-b568f50a64acc8bbe43462ffe243849a88111060b228dacb8f08d42e26985180
+    --session-entry-point "approve" --session-arg "spender:key='account-hash-9f81014b9c7406c531ebf0477132283f4eb59143d7903a2fae54358b26cea44b'"
```
Move session-arg on a new line.

ipopescu commented on source/docs/casper/workflow/erc-20/transfers.md:
```diff
@@ -0,0 +1,251 @@
+    --session-package-name "erc20_test_call"
+    --session-entry-point "check_allowance_of"
+    --session-arg "token_contract:account_hash='account-hash-b568f50a64acc8bbe43462ffe243849a88111060b228dacb8f08d42e26985180'"
+    --session-arg "owner:key='account-hash-303c0f8208220fe9a4de40e1ada1d35fdd6c678877908f01fddb2a56502d67fd'" --session-arg "spender:key='account-hash-9f81014b9c7406c531ebf0477132283f4eb59143d7903a2fae54358b26cea44b'" --chain-name integration-test
```
Move any argument with -- on a new line (here you have session-arg and chain-name).

ipopescu commented on source/docs/casper/workflow/erc-20/transfers.md:
```diff
@@ -0,0 +1,251 @@
+    --secret-key ~/casper/ibm_demo/user_b/secret_key.pem
+    --session-hash hash-b568f50a64acc8bbe43462ffe243849a88111060b228dacb8f08d42e26985180
+    --session-entry-point "transfer_from"
+    --session-arg "owner:key='account-hash-303c0f8208220fe9a4de40e1ada1d35fdd6c678877908f01fddb2a56502d67fd'" --session-arg "recipient:key='account-hash-f32a2abc55316dc85a446a1c548674e03757974aaaf86e8b7d29947ae148eeca'" --session-arg "amount:u256='10'"
```
Move the two additional session-arg to a new line.

ipopescu commented on source/docs/casper/workflow/erc-20/transfers.md:
```diff
@@ -0,0 +1,251 @@
+    --session-package-name "erc20_test_call"
+    --session-entry-point "check_balance_of"
+    --session-arg "token_contract:account_hash='account-hash-b568f50a64acc8bbe43462ffe243849a88111060b228dacb8f08d42e26985180'"
+    --session-arg "address:key='account-hash-303c0f8208220fe9a4de40e1ada1d35fdd6c678877908f01fddb2a56502d67fd'" --chain-name integration-test
```
Move chain-name to a new line.

ipopescu commented on source/docs/casper/workflow/erc-20/transfers.md:
```diff
@@ -0,0 +1,251 @@
+    --session-package-name "erc20_test_call"
+    --session-entry-point "check_balance_of"
+    --session-arg "token_contract:account_hash='account-hash-b568f50a64acc8bbe43462ffe243849a88111060b228dacb8f08d42e26985180'"
+    --session-arg "address:key='account-hash-9f81014b9c7406c531ebf0477132283f4eb59143d7903a2fae54358b26cea44b'" --chain-name integration-test
```
Move chain-name to a new line.

ipopescu commented on source/docs/casper/workflow/erc-20/transfers.md:
```diff
@@ -0,0 +1,251 @@
+    --session-package-name "erc20_test_call"
+    --session-entry-point "check_balance_of"
+    --session-arg "token_contract:account_hash='account-hash-b568f50a64acc8bbe43462ffe243849a88111060b228dacb8f08d42e26985180'"
+    --session-arg "address:key='account-hash-89422a0f291a83496e644cf02d2e3f9d6cbc5f7c877b6ba9f4ddfab8a84c2670'" --chain-name integration-test
```
Move chain-name to a new line.

ipopescu commented on source/docs/casper/workflow/erc-20/transfers.md:
```diff
@@ -0,0 +1,251 @@
+    --session-package-name "erc20_test_call"
+    --session-entry-point "check_balance_of"
+    --session-arg "token_contract:account_hash='account-hash-b568f50a64acc8bbe43462ffe243849a88111060b228dacb8f08d42e26985180'"
+    --session-arg "address:key='account-hash-f32a2abc55316dc85a446a1c548674e03757974aaaf86e8b7d29947ae148eeca'" --chain-name integration-test
```
Move chain-name to a new line.

ipopescu commented on source/docs/casper/workflow/erc-20/transfers.md:
```diff
@@ -0,0 +1,251 @@
+    --session-package-name "erc20_test_call"
+    --session-entry-point "check_allowance_of"
+    --session-arg "token_contract:account_hash='account-hash-b568f50a64acc8bbe43462ffe243849a88111060b228dacb8f08d42e26985180'"
+    --session-arg "owner:key='account-hash-303c0f8208220fe9a4de40e1ada1d35fdd6c678877908f01fddb2a56502d67fd'" --session-arg "spender:key='account-hash-9f81014b9c7406c531ebf0477132283f4eb59143d7903a2fae54358b26cea44b'" --chain-name integration-test
```
Move session-arg and chain-name to a new line.

ipopescu commented on source/docs/casper/workflow/erc-20/approvals.md:
````diff
@@ -0,0 +1,271 @@
+
+### Transfer 10 Tokens from an Allowance of only 5
+
+```
````
For all code snippets, please add the highlighting. You can do this by adding the keyword "bash" after the ```.

Example:
```bash
    --session-arg "amount:u256='10'" \
    --payment-amount "10000000000"
```

ipopescu commented on config/sidebar.config.js:
```diff
@@ -2,14 +2,28 @@ module.exports = {
+            collapsible: true,
+            collapsed: true,
+            items: [
+                "workflow/erc20/introduction",
+                "workflow/erc20/setup",
+                "workflow/erc20/query",
+                "workflow/erc20/transfers",
+                "workflow/erc20/approvals",
+                "workflow/erc20/final-balance",
```
@Buddhika-Pathirathna as you discovered, this part needs to be updated. I am adding this note here:

```
items: [
                "workflow/erc-20/introduction",
                "workflow/erc-20/setup",
                "workflow/erc-20/query",
                "workflow/erc-20/transfers",
                "workflow/erc-20/approvals",
                "workflow/erc-20/final-balance",
            ],
```

ipopescu reviewed (changes requested):
@Buddhika-Pathirathna, thank you for your work on this section! 

- Please ensure that the content matches what we have today in the old docs repository, especially the _introduction_ and _setup_ sections. Maybe we have updated some of the introductions after you started the migration (I don't remember). 
- Please use _ERC-20_ instead of _ERC20_
- Update the _sidebar.config.js_ section
- Update the code blocks to have some highlighting, and let's also discuss with the team in the style guide
- Finally, some arguments in the code examples need to be brought on a new line, for visibility

Buddhika-Pathirathna commented:
@ipopescu & @GloreenS Thank you so much for taking time for review, @ipopescu especially for checking each code block - I have addressed all the comments by you. @GloreenS - I have resolved the sidebar.config.js issue, so you must be able to load the content now.

ipopescu reviewed (approved):
@Buddhika-Pathirathna, I've noticed a couple of additional changes and I made them to speed up the merge:
- ~~Added a link to the guide on the main workflow page: https://casper.network/docs/workflow. See this: https://docs.casperlabs.io/en/latest/workflow/index.html~~ added as a mistake directly to main
- Removed the extra tabs in the code blocks. Let's discuss and hopefully keep the code blocks without extra space for easier reading and easier copy&paste.

GloreenS commented on source/docs/casper/workflow/erc-20-sample-guide/query.md:
````diff
@@ -0,0 +1,31 @@
+casper-client query-state -n http://3.143.158.19:7777
+--key hash-f9ae16d2a374e985c425c94c716c539070dc7d7907f119ba32d9684f4412a4ee
+--state-root-hash 4e8b0de303f834cb7c61bef148046e3de4446903bd15a395c9c37a6d96efe8c6
+```
````
The escape character "\" is missing at the end of each line of the command (except the last one). Same for all commands in this file.
An example of how the line break should appear:
```bash
casper-client query-state -n http://3.143.158.19:7777 \
--key hash-f9ae16d2a374e985c425c94c716c539070dc7d7907f119ba32d9684f4412a4ee \
--state-root-hash 4e8b0de303f834cb7c61bef148046e3de4446903bd15a395c9c37a6d96efe8c6 
```

GloreenS reviewed (approved):
Looks good!
A minor change to the commands listed within the files. Also, ERC20 needs to be changed to ERC-20 in the **How To's** landing page. 
screenshot attached.
<img width="459" alt="ERC-20_(10 11 2021_13 15 24_REC)" src="https://user-images.githubusercontent.com/89188883/141085543-48681ee2-28fd-4960-8802-e8ff2ec25701.png">
